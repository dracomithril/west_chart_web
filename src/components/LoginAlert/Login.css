
.btn-login a{
    margin-bottom: 0;
    margin-top: 0;
}

.login-info {
    text-align: center;
    padding: 15px;
    display: flex;
    flex-direction: column;
    margin-bottom: 40px;
    border-radius: 4px;
    color: #a94442;
    background-color: #fcf8e3;
    border: 1px solid #faebcc;
}
