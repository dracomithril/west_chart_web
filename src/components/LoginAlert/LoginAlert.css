.btn-login{
    padding: 0;
    border-radius: 4px;
    border-width: 0;
    margin-bottom: 5px;
    margin-top: 10px;
}
.btn-login a{
    margin-bottom: 0;
    margin-top: 0;
}
.login-alert {
    padding: 15px;
    margin-bottom: 40px;
    border-radius: 4px;
    color: #a94442;
    background-color: #f2dede;
    border: 1px solid #ebccd1;
}
.login-info {
    padding: 15px;
    margin-bottom: 40px;
    border-radius: 4px;
    color: #a94442;
    background-color: #fcf8e3;
    border: 1px solid #faebcc;
}
