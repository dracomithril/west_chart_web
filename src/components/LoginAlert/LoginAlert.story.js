import React from 'react';
import { addDecorator, storiesOf } from '@storybook/react';
import { Provider } from 'react-redux';
import LoginAlerts from './index';
import configureStore from '../../configureStore';

const store = configureStore();

addDecorator(story => <Provider store={store}>{story()}</Provider>);
storiesOf('LoginAlerts', module).add('Todo[VR]', () => <LoginAlerts />);
