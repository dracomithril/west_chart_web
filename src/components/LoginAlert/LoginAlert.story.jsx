import React from 'react';
import { storiesOf } from '@storybook/react';
import { Provider } from 'react-redux';
import LoginAlerts from './index';
import configureStore from '../../configureStore';

const store = configureStore();

storiesOf('LoginAlerts', module)
  .addDecorator(story => <Provider store={store}>{story()}</Provider>)
  .add('Todo[VR]', () => <LoginAlerts from="/" />);
