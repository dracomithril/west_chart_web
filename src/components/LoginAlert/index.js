export { default, Login as Component } from './Login';
