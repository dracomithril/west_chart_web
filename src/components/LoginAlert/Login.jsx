/**
 * Created by Gryzli on 28.01.2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import FacebookLogin from 'react-facebook-login/dist/facebook-login-render-props';
import Button from '@material-ui/core/Button';
import { faFacebookF, faSpotify } from '@fortawesome/free-brands-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { loginToSpotifyAlpha } from '../../utils/spotify_utils';
import './LoginAlert.css';
import { getFbPictureUrl } from '../../utils/utils';
import { api } from '../../config';
import { actionTypes } from '../../reducers/actionTypes';

const Login = ({ location }, { store }) => {
  const { user, spotifyUser } = store.getState();
  const { from } = (location || {}).state || { from: '/' };
  if (user.id && spotifyUser.id) {
    return <Redirect to={from} />;
  }
  return (
    <div className="login-info">
      <h4>
        <span role="img" aria-label="heart">
          ❤️❤️❤️
        </span>
        ️ To start working witch us you need to login to facebook and spotify.
        <span role="img" aria-label="heart">
          ❤️❤️❤️
        </span>
        ️
      </h4>
      {user.id === undefined && (
        <FacebookLogin
          appId={api.fb.apiId}
          language="pl_PL"
          autoLoad
          fields="id,email,name,first_name,last_name"
          scope="public_profile,email,groups_access_member_info,publish_to_groups"
          render={renderProps => (
            <Button
              onClick={renderProps.onClick}
              variant="raised"
              color="primary"
              style={{ height: 48, width: '100%', color: 'white' }}
            >
              <FontAwesomeIcon icon={faFacebookF} style={{ paddingRight: 5 }} />
              Login to facebook
            </Button>
          )}
          callback={(response) => {
            if (!response.error) {
              store.dispatch({
                type: actionTypes.UPDATE_USER,
                value: {
                  ...response,
                  picture_url: getFbPictureUrl(response.id),
                },
              });
              window.location = from;
            } else {
              console.error('login error.', response.error);
            }
          }}
        />
      )}
      {spotifyUser.id === undefined && (
        <Button
          variant="raised"
          style={{
            backgroundColor: 'green',
            height: 48,
            width: '100%',
            color: 'white',
          }}
          onClick={() => {
            loginToSpotifyAlpha(from)
              .then((url) => {
                window.location = url;
              })
              .catch((err) => {
                console.error(err.message);
              });
          }}
        >
          <FontAwesomeIcon icon={faSpotify} style={{ paddingRight: 5 }} />
          Login to spotify
        </Button>
      )}
    </div>
  );
};

Login.contextTypes = {
  store: PropTypes.shape(),
};
Login.propTypes = {
  location: PropTypes.shape({
    state: PropTypes.shape({
      from: PropTypes.string,
    }),
  }),
};
export default Login;
