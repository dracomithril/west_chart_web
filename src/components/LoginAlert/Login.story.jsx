import React from 'react';
import { storiesOf } from '@storybook/react';
import { Login as LoginComponent } from './Login';

storiesOf('Login', module)
  .add('Todo[VR]', () => (
    <LoginComponent
      location="/"
      autoLoad={false}
    />
  ));
