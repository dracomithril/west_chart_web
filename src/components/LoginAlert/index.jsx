/**
 * Created by Gryzli on 28.01.2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import { Jumbotron } from 'react-bootstrap';
import FacebookLogin from 'react-facebook-login';
import SpotifyLogin from './SpotifyLogin';
import './../bootstrap-social.css';
import './LoginAlert.css';
import { getFbPictureUrl } from '../../utils/utils';
import { api } from './../../config';
import { actionTypes } from './../../reducers/actionTypes';

const LoginAlert = ({ location }, { store }) => {
  const { user, spotifyUser } = store.getState();
  const { from } = (location || {}).state || { from: '/' };
  if (user.id && spotifyUser.id) {
    return <Redirect to={from} />;
  }
  return (
    <Jumbotron bsClass="login-info">
      <h4>
        <span role="img" aria-label="heart">
          ❤️❤️❤️
        </span>️ To start working witch us you need to login to facebook and spotify.
        <span role="img" aria-label="heart">
          ❤️❤️❤️
        </span>️
      </h4>
      {user.id === undefined && (
        <FacebookLogin
          appId={api.fb.apiId}
          language="pl_PL"
          autoLoad
          scope="public_profile,email,user_managed_groups"
          callback={response => {
            if (!response.error) {
              store.dispatch({
                type: actionTypes.UPDATE_USER,
                value: {
                  ...response,
                  picture_url: getFbPictureUrl(response.id),
                },
              });
            } else {
              console.error('login error.');
              console.error(response.error);
            }
          }}
          fields="id,email,name,first_name,last_name"
          cssClass="btn btn-social btn-facebook"
          icon="fab fa-facebook"
        />
      )}
      {spotifyUser.id === undefined && <SpotifyLogin from={from} />}
    </Jumbotron>
  );
};

LoginAlert.contextTypes = {
  store: PropTypes.object,
};
LoginAlert.propTypes = {
  location: PropTypes.shape({
    state: PropTypes.shape({
      from: PropTypes.string,
    }),
  }),
};
export default LoginAlert;
