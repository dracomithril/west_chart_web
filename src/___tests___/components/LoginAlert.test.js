/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import LoginAlert from './../../components/LoginAlert/index';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';

Enzyme.configure({ adapter: new Adapter() });
const initial_state = require('../data/initial_state.json');
const mockStore = configureMockStore([]);
describe('<LoginAlert/>', () => {
  beforeAll(() => {
    global.sessionStorage = jest.genMockFunction();
    global.sessionStorage.setItem = jest.genMockFunction();
    global.sessionStorage.getItem = jest.genMockFunction();
  });
  afterAll(() => {
    delete global.sessionStorage;
  });
  it('renders without crashing ChartPresenter', () => {
    const store = mockStore(initial_state);
    const wrapper = shallow(
      <LoginAlert location={{state:'some_path'}} />, {
        context: { store },
        childContextTypes: { store: PropTypes.object }
      }
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });

});
