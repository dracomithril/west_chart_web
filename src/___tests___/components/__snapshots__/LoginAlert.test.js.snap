// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginAlert/> renders without crashing ChartPresenter 1`] = `
<Jumbotron
  bsClass="login-info"
  componentClass="div"
>
  <h4>
    To start working witch us you need to login to facebook and spotify.
    <i
      className="fa fa-heart"
    />
  </h4>
  <t
    appId="1173486879387948"
    autoLoad={true}
    callback={[Function]}
    cssClass="btn btn-social btn-facebook"
    fields="id,email,name,first_name,picture,groups{administrator}"
    icon="fa fa-facebook"
    language="pl_PL"
    scope="public_profile,email,user_managed_groups"
    size="metro"
    tag="button"
    textButton="Login with Facebook"
    typeButton="button"
  />
  <SpotifyLogin />
</Jumbotron>
`;
